pkg/manager/member/tidbcluster_status_manager.go

```diff
@@ -94,13 +106,10 @@ func (m *TidbClusterStatusManager) syncDashboardMetricStorage(tc *v1alpha1.TidbC
 		return nil
 	}
 	pdEtcdClient, err := m.deps.PDControl.GetPDEtcdClient(pdapi.Namespace(tc.Namespace), tc.Name, tc.IsTLSClusterEnabled())
-
 	if err != nil {
 		return err
 	}
-	if tc.IsTLSClusterEnabled() {
```

Comment:
I have make it use the some logic in pkg/pdapi/pd_control.go

both tls enable and not enable will just return a reuse client

Reply:
Yes, but with TLS enabled, this will cause memory leak, please refer to https://github.com/pingcap/tidb-operator/issues/3058.

Reply:
we *will not* create a new client every time when calling `GetPDEtcdClient` now.
is there any reason for creating a new client every time?

Reply:
OK, sorry that I did not review the code change in that file yet. For the current behavior to create a new client each time, it's to cover the case that the secret containing the certs may be rotated to avoid expiration, in which case, the client may fail and we have to restart the tidb-controller-manager container.

Reply:
https://github.com/pingcap/tidb-operator/pull/3929/commits/9fd4f25de573f15f57385c67c192562f7e6fcb4f